src/pages/checkout.js

```diff
@@ -101,28 +106,36 @@ class Checkout extends Component {
     theme: PropTypes.object.isRequired
   };
 
+  static getDerivedStateFromProps({ cart }) {
+    if (cart.account === null && !cart.email) Router.pushRoute("login", "", { customProp: "please next" });
```

Comment:
To prevent a "flash render", caused by  the component's render being called immediately after this method is executed, this login should be move to getInitialProps https://github.com/zeit/next.js/wiki/Redirecting-in-%60getInitialProps%60